pytransform3d/transform_manager/_temporal_transform_manager.py

```diff
@@ -50,6 +54,86 @@ def as_matrix(self, time):
     def check_transforms(self):
         self._A2B = check_transform(self._A2B)
         return self
+    
+
+class PandasTimeseriesTransform(TimeVaryingTransform):
+    """Transformation, which does change over time, represented in a pandas
+    dataframe.
+
+    Parameters
+    ----------
+    df : dataframe, shape (N, 7+)
+        Time-sequence of transformations, with each row representing a single
+        sample as position-quarternion (PQ) structure.
+        Refer to `pq_column_names()` for column naming.
+
+    """
+    def __init__(self, df):
+        self._df = df
+
+    @property
+    def _time_index(self):
+        return self._df.index.to_numpy()
+    
+    def _check_if_time_is_in_bounds(self, time):
+        time_index = self._time_index
+
+        if (time < time_index[0]) or (time > time_index[-1]):
+            raise ValueError("Time value is out of range!")
+
+    def as_matrix(self, time):
+        self._check_if_time_is_in_bounds(time)
+        pq = self._interpolate_on_demand(time)
+        return transform_from_pq(pq)
+    
+    def _interpolate_on_demand(self, time):
+        # interpolate only if needed, otherwise take the corresponding
+        # sample from the dataframe
+        indices = find_neighboring_indices_around_value(self._time_index, time)
+        if indices[0] == indices[1]:
+            # Match, no interpolation needed, we just take the sample!
+            pq = self.pq_from_record(indices[0])
+        else:
+            pq = self._interpolate_pq(time)
+        return pq
+
+    def _interpolate_pq(self, time):
+        # identify the index of the preceding sample
+        t_arr = self._time_index
+        idx_timestep_earlier_wrt_query_time = np.argmax(t_arr >= time) - 1
+
+        # TODO: maybe not that efficient to do this
+        pq_array = self._df[self.pq_column_names()].to_numpy()
+        
+        # dual quaternion from preceding sample
+        t_prev = t_arr[idx_timestep_earlier_wrt_query_time]
+        pq_prev = pq_array[idx_timestep_earlier_wrt_query_time, :]
+        dq_prev = dual_quaternion_from_pq(pq_prev)
+
+        # dual quaternion from successive sample
+        t_next = t_arr[idx_timestep_earlier_wrt_query_time + 1]
+        pq_next = pq_array[idx_timestep_earlier_wrt_query_time + 1, :]
+        dq_next = dual_quaternion_from_pq(pq_next)
+
+        # since sclerp works with relative (0-1) positions
+        rel_delta_t = (time - t_prev) / (t_next - t_prev)
+        dq_interpolated = dual_quaternion_sclerp(dq_prev, dq_next, rel_delta_t)
+
+        return pq_from_dual_quaternion(dq_interpolated)
+
+    def check_transforms(self):
```

Comment:
We could also normalize quaternions here.

Reply:
Good point - should we require normalized quaternions as input?

Reply:
See my other comment

Reply:
Commented there